src/main/java/me/coley/recaf/util/ClasspathUtil.java

```diff
@@ -150,38 +155,114 @@ public void scan(ClassLoader classLoader) {
 			// auto-completion will not able to suggest internal names of any of the classes under java.*,
 			// which will largely reduce the effectiveness of the feature.
 			if (!checkBootstrapClass()) {
-
-				// The classpath for bootstrap classes can be found
-				// in the "sun.boot.class.path" property (assuming it's Oracle JVM).
-				// This was removed in Oracle JVM 9, and all related code was refactored.
-				// I use it to indicate whether the following method is supported or not.
-				if (!Strings.isNullOrEmpty(System.getProperty("sun.boot.class.path"))) {
+				float vmVersion = Float.parseFloat(System.getProperty("java.class.version")) - 44;
+				if (vmVersion < 9) {
 					try {
 						Method method = ClassLoader.class.getDeclaredMethod("getBootstrapClassPath");
 						method.setAccessible(true);
 						Field field = URLClassLoader.class.getDeclaredField("ucp");
 						field.setAccessible(true);
 
 						Object bootstrapClasspath = method.invoke(null);
-						URLClassLoader dummyLoader = new URLClassLoader(new URL[0], classLoader);
-						// Change the URLClassPath in the dummy loader to the bootstrap one.
-						field.set(dummyLoader, bootstrapClasspath);
-						// And then feed it into Guava's ClassPath scanner.
-						updateClassPath(dummyLoader);
-
-						if (!checkBootstrapClass()) {
-							Logger.warn("Bootstrap classes are (still) missing from the classpath scan!");
-						}
+						scanBootstrapClasspath(field, classLoader, bootstrapClasspath);
+						verifyScan();
 					} catch (ReflectiveOperationException | SecurityException e) {
 						throw new ExceptionInInitializerError(e);
 					}
 				} else {
-					// The internal implementation, including the class loading mechanism,
-					// was completely redesigned in Java 9. And after some hours of research,
-					// I believe that it was theoretically impossible to acquire the bootstrap classpath anymore -
-					// it seems to be completely native, and doesn't show up anywhere in the code.
-					Logger.warn("Recaf cannot acquire the classpath for bootstrap classes when running " +
-							"in Java 9 or above. This will affect auto-complete suggestions in Assembler!");
+					try {
+
+						// Before we will do that, break into Jigsaw module system to grant full access
+						Class<?> moduleClass = Class.forName("java.lang.Module");
+						Class<?> layer = Class.forName("java.lang.ModuleLayer");
+						Field lookupField = MethodHandles.Lookup.class.getDeclaredField("IMPL_LOOKUP");
+						lookupField.setAccessible(true);
+						MethodHandles.Lookup lookup = (MethodHandles.Lookup) lookupField.get(null);
+						MethodHandle export = lookup
+								.findVirtual(moduleClass, "implAddOpens", MethodType.methodType(Void.TYPE, String.class));
```

Comment:
Lines like this surpass the Checkstyle limit of 120 chars.

See results of `mvn compile`

![image](https://user-images.githubusercontent.com/21371686/62822879-9e116600-bb57-11e9-861c-5f8307c05fa4.png)

Reply:
Probably could import static methods to fix most of them.